### ISSUE TITLE
Match state as well on the `/members/mapsearch/` endpoint.

### ISSUE THREAD

**Issue by garg3133** [BitByte-TPC/alumni#126]
On the above-mentioned endpoint, currently we only match the city (we get the 'search' query as `city, state, country`). This leads to discrepancies in the profiles shown on the result page because two cities at different places can have the same name.

So, along with 'city', we should filter on the basis of the 'state' as well.

**Comment by Ritish134** [comment]
Can i take up this issue ??

**Comment by garg3133** [comment]
Sure, go ahead!

**Comment by Ritish134** [comment]
I am facing a issue that when I create an Alumni profile, it does not show up on at `/members/` endpoint and by  default there are 0 registered alumni .

So as to check the search functionality i need to create some alumni profiles ,am I missing anything as I am new to django .
Could you please help !!

**Comment by garg3133** [comment]
Also, if you're creating the users using the Registration Form on the `/newregister/` endpoint, please report any issues you come across there, as that is a new Registration Form that was just merged 2 days back.

**Comment by Ritish134** [comment]
> @Ritish134 You would need to first verify the profiles you've created by logging in as Admin, then only they'll appear on the `/members/` page. Just login with your superuser credentials (you can create them by running `python3 manage.py createsuperuser` command if not done so already), go to Admin dashboard (by clicking on the "Admin" button on the top-right corner of home page) and verify those profiles.

Yes ,did that but after verifying the profile it gives an error `NoReverseMatch at /members/`
![django_error](https://github.com/BitByte-TPC/alumni/assets/121374890/1d244c79-68b3-4dd2-8266-aff65434b5c9)
![e1](https://github.com/BitByte-TPC/alumni/assets/121374890/33487454-e8a1-4382-8614-143f623e4afe)

**Comment by garg3133** [comment]
@Ritish134 It's giving error because you've saved incorrect values in the `Batch` table. The `Batch` table stores the passing year of alumni and must be 4 digits long (ex. 2022), whereas you've set the batch for the alumni in the above screenshot as '1'.

If you look closely at the error above, there's a regex there: `[0-9]{4}` which mean that the batch should be numeric and 4 digit long, which the argument being passed is `(1,)`, i.e., '1'.

**Comment by Ritish134** [comment]
Ooh yes got it thanks 😅

**Comment by Ritish134** [comment]
How is it ?
![city-1](https://github.com/BitByte-TPC/alumni/assets/121374890/a915eb29-d6fb-4bd7-99dd-9d2eb1303e94)
![city](https://github.com/BitByte-TPC/alumni/assets/121374890/d4e76e76-de0b-404e-aad7-f297e8f09f64)

**Comment by garg3133** [comment]
@Ritish134 This isn't what I was talking about (and I should have been more clear on the steps to reproduce the issue).

This issue is related to the `/mapsearch/` endpoint. To reproduce the issue:
* Create two alumni profiles with same city names but different states (ex. Hamirpur is in both HP and UP, Aurangabad is in both Bihar and Maharashtra).
* Go to `localhost:8000/geolocation` page, zoom in to the city you just created the profile with at any one of the states, and click on it to go to the `/members/mapseach` page.
* You'll see that the list contains profiles from other states as well (but with same city name).

As for the changes you've done till now, while it looks good, we don't really need a 'state' input on the page to search for alumni. But while you're at it, maybe you can fix another issue on that page related to responsiveness.

This is how the page looks on mobile screens:

![image](https://github.com/BitByte-TPC/alumni/assets/39924567/a2680847-6b1e-40a6-ae7a-39f0d7cbac7e)

On small screens, each input should be in separate rows and wide enough for users to interact with them.

Maybe you can fix this in the current PR (as you are already working on that page) and the other issue in a separate PR?

**Comment by garg3133** [comment]
Let me know if you face any further issues here. I've filed a separate issue for the responsiveness thing: #133.
